solver/src/main.rs

```diff
@@ -169,14 +169,15 @@ async fn main() {
     .expect("failed to create gas price estimator");
 
     // TODO - Currently we use only Uniswap for price estimation.
-    let price_estimator = Arc::new(UniswapPriceEstimator::new(
+    let uniswap_pair_provider = Arc::new(UniswapPairProvider {
+        factory: contracts::UniswapV2Factory::deployed(&web3)
+            .await
+            .expect("couldn't load deployed uniswap router"),
+        chain_id,
+    });
+    let price_estimator = Arc::new(BaselinePriceEstimator::new(
         Box::new(PoolFetcher {
-            pair_provider: Arc::new(UniswapPairProvider {
-                factory: contracts::UniswapV2Factory::deployed(&web3)
-                    .await
-                    .expect("couldn't load deployed uniswap router"),
-                chain_id,
-            }),
+            pair_provider: uniswap_pair_provider,
```

Comment:
We should probably also use a filtered/cached pool fetcher here (but can be in a separate PR).

Reply:
Yes, I agree although at this moment I believe we are. I will get this PR fixed up and in order and revist if this is not the case. I would like to include a TODO in the right place so I can find and and do it immediately after this is merged.